Editor: add Length to Dynamic Array autocomplete

ericoporto opened the pull request:
fix #2599

This is working, please try.

~~this is the lowest effort for having Length in autocomplete for dynamic arrays. It looks like there are some refactoring opportunities both in the Autocomplete parser and int the typing parser in the Scintilla Wrapper, but they weren't clear for me.~~ I guess we switched for the highest effort.

~~One thing I am not doing here is adding the Length property to the call tip that shows up when the mouse over some property, I looked at the code and couldn't figure some nice way to add it there.~~ this was done

~~This adds a method `GetMembersOfDynamicArray()` that returns a hardcoded Scintilla formatted `Length` as property~~  Edit: switched for fakestruct approach.

ericoporto commented:
I switched for your branch here, I put the previous approach in other branch: https://github.com/ericoporto/ags/tree/add-length-autocomplete-bkp.

ivan-mogilko commented:
Hmm, tooltips work both in my branch and your pr for me (ah, that's practically same branch)...

But I found a silly mistake here:
```
ScriptVariable lengthVar = new ScriptVariable("Length", trueType, false, false, false, null, null, false, false, false, false, 0);
```
I made "Length" not "int" but the type of array.

ericoporto commented:
Not sure what was going on before, but the tooltip is working. I also fixed the previous mentioned issue and squashed it in the first commit, and adjusted the tests to match the new implementation.

> On a side note, that's the first time I realized that Autocomplete does not distinguish "readonly" variables, and tooltip does not mention "readonly" either.

The issue here is that there is no IsReadOnly property in the ScriptVariable class. That class has the biggest list of parameters already so I am not going to add another one now, lol, but I think the approach would be to add that information there, using `IsWordInPreviousList("readonly")` and in case of length just forcing it there through the constructor.

ericoporto commented:
I fixed the previous mentioned issue and squashed it in the first commit, and adjusted the tests to match the new implementation.

> On a side note, that's the first time I realized that Autocomplete does not distinguish "readonly" variables, and tooltip does not mention "readonly" either.

I made a new commit for this. The way it works is it shows the type name that owns the property, so it will look like this. 

![image](https://github.com/user-attachments/assets/50902608-2a68-4c69-b0a8-3dc8ca025100)

ivan-mogilko commented:
Looks like working.
